go/vt/wrangler/tablet.go

```diff
@@ -120,22 +121,21 @@ func (wr *Wrangler) DeleteTablet(ctx context.Context, tabletAlias *topodatapb.Ta
 			return err
 		}
 
-		// read the shard with the lock
-		si, err := wr.ts.GetShard(ctx, ti.Keyspace, ti.Shard)
-		if err != nil {
+		// update the shard record's master
+		echoWarning := false
+		if _, err := wr.ts.UpdateShardFields(ctx, ti.Keyspace, ti.Shard, func(s *topodatapb.Shard) error {
+			if topoproto.TabletAliasEqual(s.MasterAlias, tabletAlias) {
+				s.MasterAlias = nil
+				echoWarning = false
+			} else {
+				echoWarning = true
```

Comment:
If the master alias is not the one we're looking for, I think we should return an error from the update() func. In that case, we aren't trying to change anything in the record, so there's no reason to let UpdateShardFields proceed, and there's no reason to retry if the version happened to go bad.

In that case, you can simplify this by logging the warning here.